### PR TITLE
Fix permissions symlinks

### PR DESCRIPTION
**Pull request by cournape**
It may happen that a symlink is extracted before its target, in which case changing its mode will raise an not found file error.

### CODE REVIEW

File: zipfile2/_zipfile.py [hunk 1]
```diff
@@ -211,8 +211,7 @@ def _extract_member_to(self, member, arcname, targetpath, pwd,
             if not os.path.isdir(targetpath):
                 os.mkdir(targetpath)
             return targetpath
-
-        if is_zipinfo_symlink(member):
+        elif is_zipinfo_symlink(member):
```
Comment (itziakos) [inline zipfile2/_zipfile.py]: it looks that this function is not returning `targetpath` if this elif is executed

File: zipfile2/_zipfile.py [hunk 2]
```diff
@@ -222,17 +221,17 @@ def _extract_member_to(self, member, arcname, targetpath, pwd,
...
+                if preserve_permissions == PERMS_PRESERVE_ALL:
+                    #preserve bits 0-11: sugrwxrwxrwx, this include
+                    #sticky bit, uid bit, gid bit
+                    mode = member.external_attr >> 16 & 0xFFF
+                elif PERMS_PRESERVE_SAFE:
+                    #preserve bits 0-8 only: rwxrwxrwx
+                    mode = member.external_attr >> 16 & 0x1FF
+                os.chmod(targetpath, mode)
+
+            return targetpath
```
Comment (itziakos) [inline zipfile2/_zipfile.py]: I think that this should be moved outside the block (see https://github.com/cournape/zipfile2/pull/19/files#r34140132)

Comment (cournape) [inline zipfile2/_zipfile.py]: What is "this" here ?

Comment (itziakos) [inline zipfile2/_zipfile.py]: I was referring to `this` line given  https://github.com/cournape/zipfile2/pull/19/files#r34140132 which was fixed in the meantime.

**Comment by itziakos** [comment]
:+1:

**Pull request merged**